duckduckgo/Android
# Migrate Site Permissions UI to available components

Task/Issue URL: https://app.asana.com/0/0/1203073463571585/f

### Description
Use new design component `SectionHeaderTextView` and removed some legacy code I needed to keep in order to display existing layouts

### Steps to test this PR

_Site Permissions dialog_
- Install from this branch
- Go to www.mictest.cc
- Tap on "Test my cam" button
- [ ] Check site permissions dialog appears

_Site Permissions screen_
- Install from this branch
- Go to Settings > Site Permissions
- [ ] Check everything looks good


### No UI changes
